pkg/plugin/plugin_test.go

```diff
@@ -447,6 +448,53 @@ var _ = Describe("CNI Plugin", func() {
 				testSplitVlanIds(trunks, nil, errors.New("incorrect trunk maxID parameter"), false)
 			})
 		})
+		Context("purge ports with failed interfaces", func() {
+			const IFNAME = "eth0"
+
+			conf := fmt.Sprintf(`{
+				"cniVersion": "0.3.1",
+				"name": "mynet",
+				"type": "ovs",
+				"OvnPort": "test-port",
+				"bridge": "%s"}`, BRIDGE_NAME)
+
+			It("DEL removes ports without network namespace", func() {
+				firstTargetNs, err := testutils.NewNS()
+				Expect(err).NotTo(HaveOccurred())
+				defer firstTargetNs.Close()
+
+				secondTargetNs, err := testutils.NewNS()
+				Expect(err).NotTo(HaveOccurred())
+				defer secondTargetNs.Close()
+
+				firstResult := attach(firstTargetNs, conf, IFNAME, "", "test-port-1")
+				secondResult := attach(secondTargetNs, conf, IFNAME, "", "test-port-2")
+
+				firstHostIface := firstResult.Interfaces[0]
+				err = ip.DelLinkByName(firstHostIface.Name)
+				Expect(err).NotTo(HaveOccurred())
+
+				secondHostIface := secondResult.Interfaces[0]
+
+				args := &skel.CmdArgs{
+					ContainerID: "dummy",
+					IfName:      IFNAME,
+					StdinData:   []byte(conf),
+				}
+				err = cmdDelWithArgs(args, func() error {
+					return CmdDel(args)
+				})
+				Expect(err).NotTo(HaveOccurred())
+
+				output, err := exec.Command("ovs-vsctl", "--colum=name", "find", "Interface", fmt.Sprintf("name=%s", firstHostIface.Name)).CombinedOutput()
+				Expect(err).NotTo(HaveOccurred())
+				Expect(string(output)).To(Equal(""))
```

Comment:
We should also check that the OVS Port was deleted too, so we are not left with Interface-less Port.

Reply:
I added the check. Ports and interfaces do get removed.